test/e2e/cloudstack_test.go

```diff
@@ -4403,3 +4403,26 @@ func TestCloudstackKubernetes127To128RedHatManagementCPUpgradeAPI(t *testing.T)
 		provider.WithKubeVersionAndOS(v1alpha1.Kube128, framework.RedHat8, nil),
 	)
 }
+
+func TestCloudStackKubernetes128ValidateDomainFourLevelsSimpleFlow(t *testing.T) {
+	provider := framework.NewCloudStack(
+		t,
+		framework.WithCloudStackRedhat128(),
+		framework.WithCloudStackFillers(
+			framework.RemoveAllCloudStackAzs(),
+			framework.UpdateAddCloudStackAz4(),
+		),
+	)
+	test := framework.NewClusterE2ETest(
+		t,
+		provider,
+		framework.WithClusterFiller(
+			api.WithKubernetesVersion(v1alpha1.Kube128),
+		),
+	).WithClusterConfig(
+		api.CloudStackToConfigFiller(
+			api.WithCloudStackCredentialsRef("domain-test-creds"),
```

Comment:
Instead of passing `domain-test-creds` here, can we also get it from secret with the env variable something like T_CLOUDSTACK_CREDENTIALS_4 and pass it from the function that you have created `UpdateAddCloudStackAz4`?

Reply:
Updated